src/Validator/AddressValidator.php

```diff
@@ -34,15 +41,15 @@ public function validate($address, Constraint $constraint)
         }
 
         // Invalid postal code
-        if (!is_scalar($address->getPostalCode()) || 0 === \count(FranceCitiesBundle::getPostalCodeCities($address->getPostalCode()))) {
+        if (!is_scalar($address->getPostalCode()) || 0 === \count($this->franceCities->findCitiesByPostalCode($address->getPostalCode()))) {
             $this->context->addViolation($constraint->frenchPostalCodeMessage);
 
             return;
         }
 
         // Invalid city
         $parts = explode('-', $address->getCity());
-        if (2 !== \count($parts) || !FranceCitiesBundle::getCity($parts[0], $parts[1])) {
+        if (2 !== \count($parts) || !$this->franceCities->getCityNameByInseeCode($parts[1])) {
```

Comment:
we should validate the postal code here something like that : `in_array($parts[0], $matchedCity['postal_code'])`